pkg/test/framework/components/istio/config.go

```diff
@@ -106,6 +132,12 @@ type Config struct {
 	// Indicates that the test should deploy Istio into the target Kubernetes cluster before running tests.
 	DeployIstio bool
 
+	// Indicates that the test should deploy Istio using the alpha installer.
+	AlphaInstaller bool
+
+	// Indicates which components to install for the test. Use only if AlphaInstaller=true.
+	InstallComponents []InstallComponent
```

Comment:
There needs to be a defaulting mechanism. Otherwise these need to be repeated & specified at every call-site.

Reply:
I think there is? At https://github.com/istio/istio/blob/5e8700e69db55142feb564fe947ba81a99caba94/pkg/test/framework/components/istio/config.go#L78, the default is set to `Base` which is galley+pilot+citadel. Is there somewhere else it needs defaulting as well?

Reply:
Ah. Though, I was expecting this to mimic the "full" installation we have today. 
Do we have a case for not deploying "Base"?

Reply:
I considered it, but I think it may be better to have the minimal set and tests add the things they need. It is a bit wasteful to deploy everything when the test just uses pilot for example. 

Even if the test did something weird that cause mixer to crash or something, I don't think the test would even catch that, so there doesn't seem to be any benefit to deploying the full install if it isn't part of the test. As a benefit, they should be faster and less chance for flakes (in theory at least).

I don't know when you wouldn't want the base, but if they did they could always override it in the setup function.

If you wanted something like pilot but no galley there would be no way to do that, not sure if that is a concern